stan/math/gpu/kernels/copy_triangular_transposed_matrix_kernel.cl

```diff
@@ -0,0 +1,42 @@
+R"(
+#ifndef LOWER_TO_UPPER
+#define LOWER_TO_UPPER 1
+#endif
+
+#ifndef UPPER_TO_LOWER
+#define UPPER_TO_LOWER 0
+#endif
+
+#ifndef A
+#define A(i, j)  A[j * rows + i]
+#endif
+
+#ifndef AT
+#define AT(i, j)  A[j * cols + i]
+#endif
+
+/**
+ * Copies a lower/upper triangular of a matrix to it's upper/lower.
```

Comment:
Does this only work on square matrices? If so, perhaps indicate it in the comments here that it's assumed? I think it is because it's copying to the same `A` matrix (through the macro). Would it be clearer to just use one macro and assign it using `(j, i)` or `(i, j)` as appropriate?

Reply:
Yes, this is only meant to work on square matrices. The triangular_transpose() function in matrix_gpu checks if the matrix is square. I have added a line next to the parameter description.